amoro-common/src/main/java/org/apache/amoro/Action.java

```diff
@@ -18,87 +18,30 @@
 
 package org.apache.amoro;
 
-import org.apache.amoro.shade.guava32.com.google.common.collect.Maps;
-import org.apache.amoro.shade.thrift.org.apache.commons.lang3.tuple.Pair;
-
 import java.util.Arrays;
-import java.util.Map;
-import java.util.Set;
-import java.util.stream.Collectors;
+import java.util.Objects;
 
 public final class Action {
 
-  private static final Map<Integer, Action> ACTIONS = Maps.newConcurrentMap();
-
-  private static final TableFormat[] DEFAULT_FORMATS =
-      new TableFormat[] {TableFormat.ICEBERG, TableFormat.MIXED_ICEBERG, TableFormat.MIXED_HIVE};
-
-  static {
-    // default optimizing action
-    register(1, 10, "rewrite");
-    // expire all metadata and data files necessarily.
-    register(4, 1, "expire-data");
-    // delete orphan files
-    register(5, 2, "delete-orphans");
-    // sync optimizing commit to hive
-    register(6, 3, "sync-hive");
-  }
-
   /** supported table formats of this action */
   private final TableFormat[] formats;
-  /**
-   * storage code of this action, normally this code should be identical within supported formats
-   */
-  private final int code;
+
+  private String name;
   /**
    * the weight number of this action, the bigger the weight number, the higher positions of
    * schedulers or front pages
    */
   private final int weight;
+
   /** description of this action, will be shown in front pages */
   private final String desc;
 
-  private Action(TableFormat[] formats, int code, int weight, String desc) {
+  public Action(TableFormat[] formats, int weight, String desc) {
```

Comment:
The name is not assigned.


```suggestion
  public Action(TableFormat[] formats, int weight, String desc, String name) {
```